ceci/stage.py

```diff
@@ -34,6 +39,11 @@ class PipelineStage:
     config_options = {}
     doc = ""
 
+    inputs = []
```

Comment:
The reason I didn't have these there originally was so that the init_subclass mechanism could check to see if the user had added them correctly - does that not work in the new design?

Also it's occasionally useful to have non-final subclasses of PipelineStage, for example when grouping stages with similar functionality.

Reply:
Yeah, this was kinda an ugly point in what I did.   I need to look into this a bit more to remind myself why I did this.

Reply:
Ok, this was just to turn off a bunch of warning messages when I was linting the code / testing out the interactive stuff.  I got rid of them, but added classmethods `inputs_()` and  `outputs_()` to route as much of the access to them as a could to a place where we can provide warnings, etc... if needed.

Reply:
They seem to still be there in the web version? Do you need to push? Or is github showing me an old version (that's happened to me before!)

Reply:
The are gone for me: https://github.com/eacharles/ceci/blob/eac-dev/ceci/stage.py